greenpeace/planet4-master-theme
# PLANET-7485: Rewrite search to avoid double queries

Ref: https://jira.greenpeace.org/browse/PLANET-7485

Refactoring to:
- fix double queries
- reduce modifications of ElasticPress requests to the minimum, using filters only
- make filters and aggregation more explicit
- be compatible with ElasticPress 5.*
- work decently without ElasticPress

## Overview
```
src/Search
├── Aggregations.php				// Aggregation match against filters
├── ElasticSearch.php				// Hooks and functions related to ElasticPress/Search
├── Filters					// All search filters as displayed on search page
│   ├── ActionTypes.php
│   ├── Categories.php
│   ├── ContentTypes.php
│   ├── PostTypes.php
│   └── Tags.php
├── SearchPage.php				// Search page building, context, etc.
└── Search.php					// Core search functionality, query hooks
src/Api
└── Search.php					// Rest API for ajax request
assets/src/js
└── search.js					// Code handling the search form behavior
templates
├── search 			                // Filters templates
│   ├── action_types.twig
│   ├── categories.twig
│   ├── content_types.twig
│   └── post_types.twig
├── search.twig					// Search page template
└── tease-search.twig				// Single result template
```

## Tests
- filters
- load more
- sort
- custom results
- check [search bugs in Jira](https://jira.greenpeace.org/browse/PLANET-7363?jql=(summary%20~%20%22search*%22%20OR%20description%20~%20%22search*%22)%20AND%20type%20in%20(Bug)%20ORDER%20BY%20lastViewed%20DESC) (fixed and open)
- disable ElasticPress (optional)
